lib/bakeAmbientOcclusion.js

```diff
@@ -434,26 +437,34 @@ function computeAmbientOcclusionAt(position, normal, numberSamples, sqrtNumberSa
     for (var j = 0; j < numberSamples; j++) {
         var sampleRay = generateJitteredRay(position, normal, j, sqrtNumberSamples);
         aoBuffer.count[aoBufferIndex]++;
-        var nearestIntersect = naiveRaytrace(triangles, sampleRay, nearCull);
+        var nearestIntersect = uniformGridRaytrace(triangles, sampleRay, nearCull);
 
         if (nearestIntersect < rayDepth) {
             aoBuffer.samples[aoBufferIndex] += 1.0;
         }
     }
 }
 
-function naiveRaytrace(triangleSoup, ray, nearCull) {
-    // Check ray against every triangle in the soup. return the nearest intersection.
-    var minIntersect = Number.POSITIVE_INFINITY;
-    var triangleCount = triangleSoup.length;
-    for (var triangleSoupIndex = 0; triangleSoupIndex < triangleCount; triangleSoupIndex++) {
-        var positions = triangleSoup[triangleSoupIndex].positions;
-        var distance = rayTriangle(ray, positions[0], positions[1], positions[2], false);
-        if (defined(distance) && distance > nearCull) {
-            minIntersect = Math.min(distance, minIntersect);
-        }
+function raytraceNeighborFunction(positions, parameters) {
+    var distance = rayTriangle(parameters.ray, positions[0], positions[1], positions[2], false);
+    if (defined(distance) && distance > parameters.nearCull) {
+        parameters.minIntersect = Math.min(distance, parameters.minIntersect);
```

Comment:
I tried tweaking this on another [branch](https://github.com/likangning93/gltf-pipeline/tree/uniformGridDev). Some comparison numbers for the [2 cylinder engine model](https://github.com/KhronosGroup/glTF/tree/master/sampleModels/2CylinderEngine), which is pretty hefty:

`time node ./bin/gltf-pipeline.js ./2CylinderEngine.gltf --ao.quality=low --ao.ambientShadowContribition=1.0 --ao.rayDistance=5.0 --ao.triangleCentersOnly -o engine_AO_centers_5.gltf`
- 121496 triangles in a 149 x 55 x 54 grid
- 8 minutes and 50 seconds without the early return optimization
- 7 minutes 34 seconds with

`time node ./bin/gltf-pipeline.js ./2CylinderEngine.gltf --ao.quality=low --ao.ambientShadowContribition=1.0 --ao.rayDistance=10.0 --ao.groundPlane --ao.triangleCentersOnly -o engine_AO_centers_10_GP.gltf`
- 79 x 28 x 31 grid
- 17 minutes 13 seconds with early return
- 19 minutes 58 seconds without